src/sst/core/util/filesystem.cc

```diff
@@ -0,0 +1,274 @@
+//         size_t end = path_.find_first_of("/\\", start);
+//         part = path_.substr(start, end - start);
+//         if (part == "..") {
+//             if (!parts.empty()) {
+//                 parts.pop_back(); // Go up one directory
+//             }
+//         } else if (part != ".") {
+//             parts.push_back(part);
+//         }
+//         start = end;
+//     }
+
+//     path_ = "/" + join(parts);
+// }
+
+
+/****** Utility functions for use in the file ********/
+
+/**
+   Check to see if directory is writeable. If the directory doesn't
+   exist, false will be returned. Path must be an absolute path and if
+   it is not, function will return false.
+
+   @param path path to check
+
+   @return true if path is writable, false otherwise
+ */
+bool
+isDirectoryWritable(const std::string& path)
+{
+
+    fs::path test_file_path = path;
+    if ( !test_file_path.is_absolute() ) { return false; }
+
+    if ( !fs::is_directory(path) ) { return false; }
+
+    // Check to see if it's writable by opening a randomly named file
+    std::string test_file_name = Filesystem::getRandomName(16) + ".tmp";
+
+    test_file_path /= test_file_name;
+
+    // Attempt to create and write to the file
+    std::ofstream test_file(test_file_path.string());
+    if ( test_file ) {
+        test_file << "This is a test file." << std::endl; // Write something to the file
+        test_file.close();                                // Close the file
+        fs::remove(test_file_path);                       // Clean up by removing the file
+
+        return true; // Directory is writable
+    }
+    else {
+        return false; // Directory is not writable
+    }
+}
+
+
+/****** Class functions for Filesystem ********/
+
+bool
+Filesystem::setBasePath(const std::string& base_path)
+{
+    // Check to see if base_path exists.  If not, see if we will be
+    // able to create it when the time comes.
+    std::string absolute_path = getAbsolutePath(base_path, "");
+
+    if ( fs::is_directory(absolute_path) ) {
+        if ( isDirectoryWritable(absolute_path) ) {
+            base_path_ = absolute_path;
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    else if ( fs::exists(absolute_path) ) {
+        // This means there is a file with the same name
+        return false;
+    }
+
+    fs::path path = absolute_path;
+
+    // Directory doesn't exist.  See if we can create it.  We do this
+    // by marching back up the path until we find a path that exists,
+    // then see if we can create a directory in it.  If so, we assume
+    // we can create the whole path.
+    fs::path parent = path.parent_path();
+    while ( !fs::is_directory(parent) ) {
+        parent = parent.parent_path();
+    }
+
+    // Check to see if this existing directory is writeable
+    if ( isDirectoryWritable(parent.string()) ) {
+        base_path_ = absolute_path;
+        return true;
+    }
+
+    return false;
+}
```

Comment:
Already briefly discussed this but this seems pretty heavy-weight. We should look for a better solution for the future.